### ISSUE TITLE
Manually entered "layer name" never cleared

### ISSUE THREAD

**Issue by qgib** [qgis/QGIS#28858]
Author Name: **Jonathan Moules** (Jonathan Moules)
Original Redmine Issue: [21039](https://issues.qgis.org/issues/21039)
Affected QGIS version: 3.4.3
Redmine category:web_services_clients/wms

---
I entered a "layer name" on the WMS layer dialog (very bottom box). Now, no matter which WMS layer I select, it always has that same "layer name" showing, and every layer I add uses that QGIS layer name.

Even if I start a completely new project, it remembers my old "layer name", let along when I simply close and re-open the WMS dialog. The only way to 

The only way to clear it is either to restart QGIS entirely, or manually delete everything from the field.

I'd suggest it should be reset when a new layer is selected.

**Comment by Pedro-Murteira** [comment]
Still true on QGIS 3.16.14 and 3.22.1.

**Comment by landryb** [comment]
ok, #40421 was closed as a duplicate of this issue, but still, this behaviour is infuriating... :)

User experience:
- i load a wms server, add some layers, each with their own name provided by the server - all is fine
- at one point, only once, i need to add a new wms layer with a different/manually entered name (say, add 'prod' or 'dev' to compare layers from distinct servers, or 'jpg'/'png' to compare two renderings)
- from that point, the name i've manually entered/edited will be kept forever in the layer name field until qgis is closed, whatever the number of times i'll reopen the 'add wms layer' dialog, selecting another wms server, clicking on other layers with completely different names/content.

**Comment by landryb** [comment]
looking at the code, from my understanding of https://github.com/qgis/QGIS/blob/master/src/providers/wms/qgswmssourceselect.cpp#L989 to 'fix' this strange behaviour i'd remove `mLastLayerName` handlinh completely and directly update `leLayerName`. That would actually simplify the code, while still allowing to manually edit the layer name before adding it ? I'll test that locally.

@elpaso, since you commented on #40421, what do you think about it ?

**Comment by landryb** [comment]
```
diff --git a/src/providers/wms/qgswmssourceselect.cpp b/src/providers/wms/qgswmssourceselect.cpp
index 68e0487532..4d8da4b5b2 100644
--- a/src/providers/wms/qgswmssourceselect.cpp
+++ b/src/providers/wms/qgswmssourceselect.cpp
@@ -986,30 +986,28 @@ void QgsWMSSourceSelect::updateButtons()
     }
   }
 
-  if ( leLayerName->text().isEmpty() || leLayerName->text() == mLastLayerName )
+  if ( leLayerName->text().isEmpty() )
   {
     if ( addButton()->isEnabled() )
     {
       if ( !lstTilesets->selectedItems().isEmpty() )
       {
         QTableWidgetItem *item = lstTilesets->selectedItems().first();
-        mLastLayerName = item->data( Qt::UserRole + 5 ).toString();
-        if ( mLastLayerName.isEmpty() )
-          mLastLayerName = item->data( Qt::UserRole + 0 ).toString();
-        leLayerName->setText( mLastLayerName );
+        QString tileLayerName = item->data( Qt::UserRole + 5 ).toString();
+        if ( tileLayerName.isEmpty() )
+          tileLayerName = item->data( Qt::UserRole + 0 ).toString();
+        leLayerName->setText( tileLayerName );
       }
       else
       {
         QStringList layers, styles, titles;
         collectSelectedLayers( layers, styles, titles );
-        mLastLayerName = titles.join( QLatin1Char( '/' ) );
-        leLayerName->setText( mLastLayerName );
+        leLayerName->setText( titles.join( QLatin1Char( '/' ) );
       }
     }
     else
     {
-      mLastLayerName.clear();
-      leLayerName->setText( mLastLayerName );
+      leLayerName->setText( "" );
     }
   }
 }
diff --git a/src/providers/wms/qgswmssourceselect.h b/src/providers/wms/qgswmssourceselect.h
index 739a5d18af..12604f79a4 100644
--- a/src/providers/wms/qgswmssourceselect.h
+++ b/src/providers/wms/qgswmssourceselect.h
@@ -160,9 +160,6 @@ class QgsWMSSourceSelect : public QgsAbstractDataSourceWidget, private Ui::QgsWM
     //! URI for selected connection
     QgsDataSourceUri mUri;
 
-    //! layer name derived from latest layer selection (updated as long it's not edited manually)
-    QString mLastLayerName;
-
     //! The widget that controls the image format radio buttons
     QButtonGroup *mImageFormatGroup = nullptr;
 
```

a bit crude but you get the idea.

**Comment by gioman** [comment]
> a bit crude but you get the idea.

@landryb better post the patch as a Pull Request, otherwise is likely no one will ever review it.

**Comment by landryb** [comment]
sure, i'll do a PR once i've locally tested a patch :)
